include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,766 @@
+            if (lhs[i] != rhs[i])
+                return false;
+
+        return true;
+        // return ranges::equal(lhs, rhs); // does not work
+    }
+
+    //!\copydoc operator==
+    constexpr friend bool operator!=(gap_decorator_anchor_set const & lhs, gap_decorator_anchor_set const & rhs) noexcept
+    {
+        return !(lhs == rhs);
+    }
+
+    constexpr friend bool operator<(gap_decorator_anchor_set const & lhs, gap_decorator_anchor_set const & rhs) noexcept
+    {
+        size_t i{0};
+        while (i < std::min<size_t>(lhs.size(), rhs.size()) && lhs[i] == rhs[i])
+            ++i;
+
+        if (i == lhs.size())
+        {
+            if (i == rhs.size())  // lhs == rhs
+                return false;
+            else
+                return true;          // lhs prefix of rhs
+        }
+        else if (i == rhs.size())
+        {
+            return false;             // rhs prefix of lhs
+        }
+
+        return lhs[i] < rhs[i];
+    }
+
+    constexpr friend bool operator<=(gap_decorator_anchor_set const & lhs, gap_decorator_anchor_set const & rhs) noexcept
+    {
+        size_t i{0};
+        while (i < std::min<size_t>(lhs.size(), rhs.size()) && lhs[i] == rhs[i])
+            ++i;
+
+        if (i == lhs.size())
+        {
+            return true;          // lhs == rhs, or lhs prefix of rhs
+        }
+        else if (i == rhs.size())
+        {
+            return false;             // rhs prefix of lhs
+        }
+
+        return lhs[i] < rhs[i];
+    }
+
+    constexpr friend bool operator>(gap_decorator_anchor_set const & lhs, gap_decorator_anchor_set const & rhs) noexcept
+    {
+        return !(lhs <= rhs);
+    }
+
+    constexpr friend bool operator>=(gap_decorator_anchor_set const & lhs, gap_decorator_anchor_set const & rhs) noexcept
+    {
+        return !(lhs < rhs);
+    }
+    //!\}
+
+private:
+    //!\brief The gap type as a tuple storing position and accumulated gap lengths.
+    using anchor_gap_t = typename std::pair<size_t, size_t>;
+
+    //!\brief The type of set to store the anchor gaps.
+    using anchor_set_type = std::set<anchor_gap_t>;
+
+    //!\brief The iterator type for an anchor set.
+    using set_iterator_type = typename anchor_set_type::iterator;
+
+    //!\brief The maximum value is needed for a correct search with upper_bound() in the anchor set.
+    inline static size_t const bound_dummy{std::numeric_limits<size_t>::max()};
+
+    /*!\brief Helper function to compute the length of the gap indicated by the
+     * input iterator.
+     * \param[in] it    Iterator over the internal gap set.
+     * \returns The gap length corresponding to the gap pointed at by \p it.
+     *
+     * \details
+     *
+     * The length of a gap is difference of the accumulator pointed at
+     * and the one of its predecessor (if existing).
+     *
+     * ### Exceptions
+     * No-throw guarantee.
+     */
+    constexpr size_type get_gap_length(set_iterator_type it) const noexcept
```

Comment:
I would recommend to use `get` and `set` only for getters and setters, not for other function names.